yagna_erigon_manager.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+from yapapi.log import enable_default_logger, log_summary, log_event_repr
+from uuid import uuid4
+from turbogeth_payload import TurbogethPayload
+from yapapi import Executor, Task
+from datetime import timedelta
+from worker import worker
+
+SUBNET_TAG = 'ttt'
+
+
+class Erigon():
+    def __init__(self):
+        self.id = self._create_id()
+        self.queue = asyncio.Queue()
+
+    async def start(self):
+        fut = asyncio.get_running_loop().create_future()
+        await self.queue.put(fut)
+        await fut
+
+    async def status(self):
+        return await self.run('STATUS')
+
+    async def stop(self):
+        return await self.run('STOP')
+
+    async def run(self, cmd):
+        fut = asyncio.get_running_loop().create_future()
+        await self.queue.put((cmd, fut))
+        await fut
+        return fut.result()
+
+    def _create_id(self):
+        return uuid4().hex
+
+    def __repr__(self):
+        return f"{type(self).__name__}[id={self.id}]"
+
+
+class YagnaErigonManager():
+    def __init__(self):
+        enable_default_logger(log_file='log.log')
+        self.executor_task = None
+        self.tasks_queue = asyncio.Queue()
+        self.closing = False
+
+    async def deploy_erigon(self):
+        if self.executor_task is None:
+            self.executor_task = asyncio.create_task(self._create_executor())
+
+        erigon = Erigon()
+        self.tasks_queue.put_nowait(erigon)
+        await erigon.start()
```

Comment:
Start should happen outside of this function so that we're able to return quickly.

Reply:
ce1ae67